src/Provider/TMSProvider.js

```diff
@@ -23,32 +24,56 @@ function preprocessDataLayer(layer) {
             max: 18,
         };
     }
+    layer.fx = layer.fx || 0.0;
+}
+
+function getVectorTile(tile, coords, layer) {
+    const url = URLBuilder.xyz(coords, layer);
+
+    if (layer.type == 'color') {
+        return VectorTileHelper.getVectorTileTextureByUrl(url, tile, layer, coords)
+            .then((texture) => {
+                const result = {
+                    texture,
+                };
+                return result;
+            });
+    }
 }
 
 function executeCommand(command) {
     const layer = command.layer;
     const tile = command.requester;
 
     const promises = [];
+    const supportedFormats = {
+        'application/x-protobuf;type=mapbox-vector': getVectorTile.bind(this),
+    };
+    const func = supportedFormats[layer.format];
+
     for (const coordTMS of tile.getCoordsForLayer(layer)) {
         const coordTMSParent = (command.targetLevel < coordTMS.zoom) ?
             OGCWebServiceHelper.WMTS_WGS84Parent(coordTMS, command.targetLevel) :
             undefined;
```

Comment:
Is it intentionnal that `coordTMSParent` is not used for vector tiles ? if yes, please move it to the OGC case.
My guess is that `const urld = URLBuilder.xyz(coordTMSParent || coordTMS, layer);` could be instead moved out here of the format switch (and of `getVectorTile`).

Reply:
Yes it is intentional. What do you mean by moving it to the OGC case ?

Reply:
`URLBuilder.xyz` is called at two places with the same arguments : 
- L31 `const url = URLBuilder.xyz(coords, layer); ` 
- L55 `const urld = URLBuilder.xyz(coordTMSParent || coordTMS, layer); `
You could move it right after L51 and pass the url to `VectorTileHelper.getVectorTileTextureByUrl` directly (removing the need for a `getVectorTile` function)

Reply:
Yeah I forgot to remove the `coordTMSParent ||` for the vector tile when I was testing before, if it is added tiles are not showing correctly.

Reply:
Does it make sense to support `command.targetLevel` and `coordTMSParent` for vector tiles as well ? if not, why ? if yes, then I think this feature could be added to Vector tiles as well, that would help to have an equal treatment for OGCWebService, VectorTiles and any future formats... 

It is not a blocker to me, but if you merge it like this please open an issue about this for future reference.